src/Microsoft.ML.TensorFlow/TensorFlow/TensorflowUtils.cs

```diff
@@ -158,6 +160,124 @@ internal static TFSession LoadTFSession(IExceptionContext ectx, byte[] modelByte
             return new TFSession(graph);
         }
 
+        private static TFSession LoadTFSession(IHostEnvironment env, string exportDirSavedModel)
+        {
+            env.CheckValue(exportDirSavedModel, nameof(exportDirSavedModel));
+            var sessionOptions = new TFSessionOptions();
+            var exportDir = exportDirSavedModel;
+            var tags = new string[] { "serve" };
+            var graph = new TFGraph();
+            var metaGraphDef = new TFBuffer();
+
+            return TFSession.FromSavedModel(sessionOptions, null, exportDir, tags, graph, metaGraphDef);
+        }
+
+        // A TensorFlow frozen model is a single file. An un-frozen (SavedModel) on the other hand has a well-defined folder structure.
+        // Given a modelPath, this utility method determines if we should treat it as a SavedModel or not
+        internal static bool IsSavedModel(IHostEnvironment env, string modelPath)
+        {
+            env.CheckNonWhiteSpace(modelPath, nameof(modelPath));
+            FileAttributes attr = File.GetAttributes(modelPath);
+            return attr.HasFlag(FileAttributes.Directory);
+        }
+
+        // Currently used in TensorFlowTransform to protect temporary folders used when working with TensorFlow's SavedModel format.
+        // Models are considered executable code, so we need to ACL tthe temp folders for high-rights process (so low-rights process can’t access it).
+        /// <summary>
+        ///  Given a folder path, create it with proper ACL if it doesn't exist.
+        ///  Fails if the folder name is empty, or can't create the folder.
+        /// </summary>
+        internal static void CreateFolderWithAclIfNotExists(IHostEnvironment env, string folder)
```

Comment:
>CreateFolderWithAclIfNotExists [](start = 29, length = 30)

How does creation of folder ACL work in Linux/Mac?

Reply:
Catches the PlatformNotSupported exception and does the normal create.

---
In reply to: [219903672](https://github.com/dotnet/machinelearning/pull/853#discussion_r219903672) [](ancestors = 219903672)

Reply:
Thanks John, got it. 

---
In reply to: [219906478](https://github.com/dotnet/machinelearning/pull/853#discussion_r219906478) [](ancestors = 219906478,219903672)